libpod/container_api.go

```diff
@@ -689,15 +689,22 @@ func (c *Container) Wait() (int32, error) {
 	return exitCode, nil
 }
 
-// CleanupStorage unmounts all mount points in container and cleans up container storage
-func (c *Container) CleanupStorage() error {
+// Cleanup unmounts all mount points in container and cleans up container storage
+// It also cleans up the network stack
+func (c *Container) Cleanup() error {
 	if !c.locked {
 		c.lock.Lock()
 		defer c.lock.Unlock()
 		if err := c.syncContainer(); err != nil {
 			return err
 		}
 	}
+
+	// Stop the container's network namespace (if it has one)
+	if err := c.runtime.teardownNetNS(c); err != nil {
```

Comment:
Need to also clear a few things in the DB after this - c.state.NetNS, c.state.IPAddress, c.state.SubnetMask. And probably need to do a c.save() afterwards to get that into the DB.

Reply:
Fixed